os_stub/spdm_device_secret_lib_sample/lib.c

```diff
@@ -28,6 +28,789 @@
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        result = libspdm_rsa_set_key(rsa_context, LIBSPDM_RSA_KEY_E, m_libspdm_rsa2048_req_e,
+                                     sizeof(m_libspdm_rsa2048_req_e));
+        if (!result) {
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        result = libspdm_rsa_set_key(rsa_context, LIBSPDM_RSA_KEY_D, m_libspdm_rsa2048_req_d,
+                                     sizeof(m_libspdm_rsa2048_req_d));
+        if (!result) {
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        *context = rsa_context;
+        return true;
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSASSA_3072:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSAPSS_3072:
+        rsa_context = libspdm_rsa_new();
+        if (rsa_context == NULL) {
+            return false;
+        }
+        result = libspdm_rsa_set_key(rsa_context, LIBSPDM_RSA_KEY_N, m_libspdm_rsa3072_req_n,
+                                     sizeof(m_libspdm_rsa3072_req_n));
+        if (!result) {
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        result = libspdm_rsa_set_key(rsa_context, LIBSPDM_RSA_KEY_E, m_libspdm_rsa3072_req_e,
+                                     sizeof(m_libspdm_rsa3072_req_e));
+        if (!result) {
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        result = libspdm_rsa_set_key(rsa_context, LIBSPDM_RSA_KEY_D, m_libspdm_rsa3072_req_d,
+                                     sizeof(m_libspdm_rsa3072_req_d));
+        if (!result) {
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        *context = rsa_context;
+        return true;
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSASSA_4096:
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSAPSS_4096:
+        rsa_context = libspdm_rsa_new();
+        if (rsa_context == NULL) {
+            return false;
+        }
+        result = libspdm_rsa_set_key(rsa_context, LIBSPDM_RSA_KEY_N, m_libspdm_rsa4096_req_n,
+                                     sizeof(m_libspdm_rsa4096_req_n));
+        if (!result) {
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        result = libspdm_rsa_set_key(rsa_context, LIBSPDM_RSA_KEY_E, m_libspdm_rsa4096_req_e,
+                                     sizeof(m_libspdm_rsa4096_req_e));
+        if (!result) {
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        result = libspdm_rsa_set_key(rsa_context, LIBSPDM_RSA_KEY_D, m_libspdm_rsa4096_req_d,
+                                     sizeof(m_libspdm_rsa4096_req_d));
+        if (!result) {
+            libspdm_rsa_free(rsa_context);
+            return false;
+        }
+        *context = rsa_context;
+        return true;
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P256:
```

Comment:
The logic for RSA are same. Please merge them into one common case.

The only difference is the key, you can use a sub switch for them.